src/lib/marshal/email-layout/helpers.ts

```diff
@@ -22,3 +30,95 @@ export const lsEmailLayoutJson = async (
   const exists = await fs.pathExists(emailLayoutJsonPath);
   return exists ? emailLayoutJsonPath : undefined;
 };
+
+/*
+ * Validate the provided args and flags with the current run context, to first
+ * ensure the invoked command makes sense, and return the target context.
+ */
+type CommandTargetProps = {
+  flags: {
+    all: boolean | undefined;
+    "layouts-dir": DirContext | undefined;
+  };
+  args: {
+    emailLayoutKey: string | undefined;
+  };
+};
+type LayoutDirTarget = {
+  type: "layoutDir";
+  context: EmailLayoutDirContext;
+};
+type LayoutsIndexDirTarget = {
+  type: "layoutsIndexDir";
+  context: DirContext;
+};
+
+export type LayoutCommandTarget = LayoutDirTarget | LayoutsIndexDirTarget;
```

Comment:
```suggestion
export type EmailLayoutCommandTarget = EmailLayoutDirTarget | EmailLayoutsIndexDirTarget;
```
Sorry to get hung up on naming, but I think it'd be good to consistently refer to "email layout" in code / internally - especially for defining type / function / file names etc (that are more "official" and will stick around).

We can use the shortened "layout" term for user facing messages and commands. Ans for short lived var names etc is probably fine too I think.

Reply:
Yes you are right! https://github.com/knocklabs/knock-cli/pull/236/commits/85bd99997b795acc0c2d0758746f533cf640756a